ocs_ci/ocs/ocs_upgrade.py

```diff
@@ -385,3 +391,132 @@ def set_upgrade_images(self):
             with NamedTemporaryFile() as cs_yaml:
                 dump_data_to_temp_yaml(cs_data, cs_yaml.name)
                 ocs_catalog.apply(cs_yaml.name)
+
+
+class Tier4aUpgrade(EcosystemTest):
+
+    def get_ocs_operator_node_name(self):
+        """
+        Getting node's name that running ocs-operator pod
+
+        Returns:
+            str: node's name that running ocs-operator pod
+
+        """
+        ocs_operator_pod = get_ocs_operator_pod()
+        log.debug(f"ocs operator pod info: {ocs_operator_pod}")
+        ocs_operator_node = get_pod_node(ocs_operator_pod)
+
+        return get_node_name(ocs_operator_node)
+
+    def worker_node_shutdown(self, nodes, abrupt):
+        """
+        Shutdown worker node that running ocs-operator
+
+        """
+        log.info(f"Abrupt {abrupt}")
+        # get ocs-operator node:
+        ocs_operator_node_name = self.get_ocs_operator_node_name()
+        # get rook-ceph-tools node:
+        rook_ceph_tools_pod = get_ceph_tools_pod()
+        rook_ct_node = get_pod_node(rook_ceph_tools_pod)
+        rook_ct_node_name = get_node_name(rook_ct_node)
+        if rook_ct_node_name == ocs_operator_node_name:
+            delete_pods([get_ocs_operator_pod()])
+            time.sleep(120)
+            ocs_operator_node_name = self.get_ocs_operator_node_name()
+            assert not (ocs_operator_node_name == rook_ct_node_name),\
+                "rook-ceph-tools and ocs-operator pods running on same node"
+        # get workers node objects:
+        node_to_shutdown = list()
+        for node in get_typed_nodes():
+            node_name = get_node_name(node)
+            log.info(f"node: {node_name}, ocs operator node: {ocs_operator_node_name}")
+            if node_name == ocs_operator_node_name:
+                if node_name != rook_ct_node_name:
+                    node_to_shutdown.append(node)
+                    log.info(f"node to shutdown: {get_node_name(node_to_shutdown[0])}")
+                    nodes.stop_nodes(node_to_shutdown)
+                    log.info(f"stop instance - done!")
+                    break
+
+        log.info("Sleeping 5 minutes")
+        time.sleep(300)
```

Comment:
it should be depends on how we want the test to be designed. sleep here give oc time to move components from node that shut down to another node. @nehaberry , what do u think ?

Reply:
so let's poll these components movement to the other node

Reply:
@AviadP +1 IMO too.. 

In case of node shutdown : The default k8s timeout before pods move to another node is 5 mins... So till the node is OFF for 5 mins, all the pods will stay in the same state. Hence, we might need to wait for minimum 5 mins before we observe any change in the cluster.

if we poll within this 5 mins.. all the pods would already be showing as Running.. So we need to handle it properly .

Reply:
with @nehaberry comment, will leave it without polling. and increase timeout by 20 sec, to be on safer side